test_tracetools/CMakeLists.txt

```diff
@@ -1,9 +1,10 @@
 cmake_minimum_required(VERSION 3.5)
 project(test_tracetools)
 
-# Default to C++14
+# Default to C++17
 if(NOT CMAKE_CXX_STANDARD)
-  set(CMAKE_CXX_STANDARD 14)
+  set(CMAKE_CXX_STANDARD 17)
+  set(CMAKE_CXX_STANDARD_REQUIRED ON)
```

Comment:
I'm guessing making the same changes in `tracetools` isn't strictly necessary? It would probably be a good idea, though.

Reply:
Oh, you know, it didn't come up in my compilation tests with clang, so I didn't include it.  But I agree that we should just make it consistent, so I'll update this PR with that.

Reply:
Done in 3d3a885